src/crimson/os/seastore/cache.cc

```diff
@@ -696,6 +696,13 @@ void Cache::add_extent(CachedExtentRef ref)
   }
 }
 
+void Cache::add_extent_no_cache(CachedExtentRef ref)
```

Comment:
This name is confusing -- ref does get added to the cache.  Instead, let's add an optional t_src param to add_extent and update add_extent to:

```
void Cache::add_extent(
  CachedExtentRef ref,
  const Transaction::src_t* p_src=nullptr)
{
  assert(ref->is_valid());
  extents.insert(*ref);
  if (ref->is_dirty()) {
    add_to_dirty(ref);
  } else if (p_src && !is_cleaner_transaction(t.get_src()) {
    touch_extent(*ref);
  }
}
```

Reply:
Or not optional if all users have it?

Reply:
ok

Reply:
all users have it except in `cache::replay_delta` while there is not a transaction. I set this param as Transaction::src_t::READ in this situation, i think it might be proper?

Reply:
Seems ok

Reply:
> all users have it except in `cache::replay_delta` while there is not a transaction. I set this param as Transaction::src_t::READ in this situation, i think it might be proper?

will this be counted in Transaction READ metrics? if yes, may cause it not accurate.  if not then no problem.

Reply:
> will this be counted in Transaction READ metrics? if yes, may cause it not accurate. if not then no problem.

it seems not

Reply:
Agree with chunmei that this is not accurate as it is not a READ transaction.